Userland/Libraries/LibPDF/Filter.cpp

```diff
@@ -175,23 +175,79 @@ PDFErrorOr<ByteBuffer> Filter::decode_png_prediction(Bytes bytes, size_t bytes_p
     return decoded;
 }
 
+PDFErrorOr<ByteBuffer> Filter::decode_tiff_prediction(Bytes bytes, int columns, int colors, int bits_per_component)
+{
+    // TIFF 6 spec, Section 14: Differencing Predictor.
+    // "Assuming 8-bit grayscale pixels for the moment, a basic C implementation might look something like this:
+    //     char image[][]; int row, col;
+    //     /* take horizontal differences: */
+    //     for (row = 0; row < nrows; row++)
+    //         for (col = ncols - 1; col >= 1; col--)
+    //             image[row][col] -= image[row][col-1];
+    //  If we don’t have 8-bit components, we need to work a little harder [...]
+    //  Suppose we have 4-bit components packed two per byte in the normal TIFF uncompressed [...]
+    //  first expand each 4-bit component into an 8-bit byte, so that we have one component per byte, low-order justified.
+    //  We then perform the horizontal differencing [...] we then repack the 4-bit differences two to a byte [...]
+    //  If the components are greater than 8 bits deep, expanding the components into 16-bit words instead of 8-bit bytes
+    //  seems like the best way to perform the subtraction on most computers. [...]
+    //  An additional consideration arises in the case of color images. [...]
+    //  we will do our horizontal differences with an offset of SamplesPerPixel [...
+    //  In other words, we will subtract red from red, green from green, and blue from blue"
+    //
+    // The spec text describes the forward transform; this implements the inverse.
+    //
+    // Section 8: Baseline Field Reference Guide, entry for Compression:
+    // "Each row is padded to the next BYTE/SHORT/LONG boundary, consistent with
+    //  the preceding BitsPerSample rule."
```

Comment:
I'm quite split on that because half of me is happy to see spec comments, but I often find the TIFF spec more confusing than helpful.
So copying a whole block of it doesn't seem really necessary, and I would prefer something like:
"It's encoded as a difference between each pixel, while doing the operation on each channel separately (trust me bro)
This described in Section 14: Differencing Predictor (good luck)"

Reply:
I like literal quotes from specs, with elision as needed. That's also inline with serenity style, so I'll keep this. (I agree the TIFF spec is very wordy.)

Reply:
Fair enough!